private/bufpkg/bufimage/bufimageutil/bufimageutil.go

```diff
@@ -452,12 +453,21 @@ func (t *transitiveClosure) addElement(
 			if !isIncluded {
 				continue
 			}
+			if index := field.OneofIndex; index != nil {
+				oneofFieldCounts[*index]++
```

Comment:
Might be worth a range check to avoid a possible panic. This would be useful int the case of someone feeding the image as a binary proto (i.e. like the result of `buf bulid`), but was passing an invalid or corrupted schema.

Reply:
Done [d327c23](https://github.com/bufbuild/buf/pull/3624/commits/d327c23f59e2ebc2e6ee9c58510ab629d190cbef)